### DISCUSSION TITLE
config.load_incluster_config() while developing

### DISCUSSION THREAD

**Discussion by agardnerIT** [nolar/kopf#1050] [Q&A] [answered]
Hi all,
I'm writing an Operator with kopf (kopf is incredible btw!). I want to develop locally using `kopf run my-operator.py` AND then package that file for use in_cluster.

Obviously, using "live" (in cluster), I need:

```
config.load_incluster_config()
```

But that errors when I run locally.

Has anyone come up with a solution?

Edit: I must be tired. A try / except of course:

```
    try:
        configuration = config.load_incluster_config()   
    except Exception:
        logger.info("Cannot load config from within cluster. Loading local config. This is OK if you are a developer.")
        configuation = config.load_config()
```

**Comment by nolar** [comment, marked as answer]
Depends on what you are trying to authenticate — Kopf itself or your own client object. If the latter, it is better to address this question to that library's communities.

If the former, Kopf already authenticates with in-cluster/local configs if either the official client or pykube-ng are installed, i.e. it works "out of the box" in both environments: 

* Docs: https://kopf.readthedocs.io/en/stable/authentication/#piggybacking
* Official client auth: https://github.com/nolar/kopf/blob/538df59b88d1aab7b985d703483497f73c6c4783/kopf/_core/intents/piggybacking.py#L50
* Pykube-ng auth: https://github.com/nolar/kopf/blob/538df59b88d1aab7b985d703483497f73c6c4783/kopf/_core/intents/piggybacking.py#L105
